module/VuFind/src/VuFind/ILS/Driver/Demo.php

```diff
@@ -1663,13 +1667,16 @@ public function cancelStorageRetrievalRequests($cancelDetails)
      * separated by a pipe, which is then submitted as form data in Hold.php. This
      * value is then extracted by the CancelHolds function.
      *
-     * @param array $details An array of item data
+     * @param array $request An array of request data
+     * @param array $patron  Patron information from patronLogin
      *
      * @return string Data for use in a form field
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function getCancelStorageRetrievalRequestDetails($details)
+    public function getCancelStorageRetrievalRequestDetails($request, $patron)
```

Comment:
Should getCancelStorageRetrievalRequestDetails have a default value for $patron for consistency?

Reply:
I don't think so. It has always had the patron parameter, and it doesn't make sense to call it unless there are requests to display and thus a known patron.

Reply:
I don't think it's true that this has always had the patron parameter, because you're adding the patron parameter here. However, I agree with your point that it isn't necessary to make $patron optional in this context. :-)

Reply:
Actually, though, it does seem to be a problem that there are several instances of getCancelStorageRetrievalRequestDetails that have not had the patron parameter added yet in this PR (KohaRest, for example).

Reply:
Sorry, I wasn't quite clear: The parameter has always been passed to getCancelStorageRetrievalRequestDetails even if it hasn't been declared for all the instances. See https://github.com/vufind-org/vufind/commit/3bec0eb7d3cb9971fe446cf00f14c7f32348d09a#diff-c2869125639e00db24fef78a00af0fa67f1d25a7770931b86c0118902437ef77R85

Reply:
Ahh, I see, that makes sense. I still think it's worth making these method signatures consistent across drivers, in case we implement feature interfaces in the future (and just to prevent potential confusion). I've gone ahead and added $patron to the drivers that were missing it. (See commit f6d1968707c94103d8c8ce61b78f26a282cf2151).